### PR TITLE
ARROW-7048: [Java] Support for combining multiple vectors under VectorSchemaRoot

### PR DESCRIPTION
**Pull request by liyafan82**
For details, please see https://issues.apache.org/jira/browse/ARROW-7048

### CODE REVIEW

**Comment by emkornfield** [comment]
@TheNeuralBit do you have time to review?

File: java/vector/src/main/java/org/apache/arrow/vector/util/VectorSchemaRootAppender.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+   */
+  public static void append(boolean checkSchema, VectorSchemaRoot targetRoot, VectorSchemaRoot... rootsToAppend) {
+    // create appenders
+    VectorAppender[] appenders = new VectorAppender[targetRoot.getFieldVectors().size()];
+    for (int i = 0; i < appenders.length; i++) {
+      appenders[i] = new VectorAppender(targetRoot.getVector(i));
+    }
+
+    // create type checkers, if necessary
+    TypeEqualsVisitor[] typeCheckers = new TypeEqualsVisitor[targetRoot.getFieldVectors().size()];
```
Comment (emkornfield) [inline java/vector/src/main/java/org/apache/arrow/vector/util/VectorSchemaRootAppender.java]: maybe only allocate this is checkSchema = true?

**Comment by emkornfield** [comment]
LGTM, one small comment, thanks @liyafan82

File: java/vector/src/main/java/org/apache/arrow/vector/util/VectorSchemaRootAppender.java (hunk 1 above)
Comment (liyafan82) [reply java/vector/src/main/java/org/apache/arrow/vector/util/VectorSchemaRootAppender.java]: Revised. Thank you for the good suggestion.

**Pull request closed without merging**